### PR TITLE
Document TRL and DSOS

### PR DESCRIPTION
**Pull request by mlenkeit**
This PR aims at fixing #832.

### CODE REVIEW

**Comment by Ein-Tim** [comment]
What is blocking this PR from being merged?

**Comment by MikeMcC399** [comment]
@Ein-Tim 
Currently the PR is a collection of images which are not embedded anywhere in a document, so it is incomplete. Perhaps @mlenkeit can let us know his plans?

**Comment by MikeMcC399** [comment]
@mlenkeit 
Are you going to continue with this PR sometime?

**Comment by mlenkeit** [comment]
> @mlenkeit Are you going to continue with this PR sometime?

yes.

**Comment by mlenkeit** [comment]
I have reworked the chapters on the risk calculation to make the role of TRL and DSOS and the respective mapping a bit clearer.

@MikeMcC399 I would appreciate your feedback if you find some time.

File: solution_architecture.md [hunk 1]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
+    <details>
+    <summary>Figure 17: Determine DSOS for 'No information'</summary>
+
+    | ![Figure 17: Determine DSOS for 'No information'](images/solution_architecture/dsos-no-information.png "Figure 17: Determine DSOS for 'No information'") |
+    |:--:|
+    | **Figure 17: Determine DSOS for 'No information'**|
+
+    </details>
+
+2. **No symptoms** - the user indicates that he/she has symptoms
```
Comment (MikeMcC399) [inline solution_architecture.md:306]: "the user indicates that he/she has symptoms" =>

"the user indicates that he/she has no symptoms" ?

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Please check the comment about no symptoms.

**Comment by MikeMcC399** [comment]
@mlenkeit 

> I have reworked the chapters on the risk calculation to make the role of TRL and DSOS and the respective mapping a bit clearer.

Many thanks!

> @MikeMcC399 I would appreciate your feedback. I'll leave the PR open for a few days.

I only did a brief sanity check on the changes and I have just one minor comment to take a look at. Otherwise it looks good to merge.

File: solution_architecture.md (hunk 1 above)
Comment (mlenkeit) [reply solution_architecture.md]: Thanks, good catch (as usual) 👍 Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/b1505f1cb5a031acd2586d89ec567282a2649c23.

**Review by MikeMcC399 — APPROVED** [review]
LGTM!

File: solution_architecture.md [hunk 2]
```diff
@@ -225,24 +229,24 @@ For Apple devices an OS version of at least 12.5 (for older devices) or 13.7 is
 
 For Android devices, the features are integrated into the [Google Play Services](https://developers.google.com/android/exposure-notifications/exposure-notifications-api#architecture), which means that only this specific application needs to be updated for it to work. Devices starting with Android 6.0 (API version 23) and integrated BLE chips are [supported](https://developers.google.com/android/exposure-notifications/exposure-notifications-api#architecture).
 
+### ENF Usage
+
+The Exposure Notification Framework (ENF) encapsulates handling of the keys, including all cryptographic operations on them, and broadcasting and scanning for Bluetooth advertisement packages for the app (see *Figure 11* and *Figure 12*). The app itself does not have access to the collected exposures, i.e. the Rolling Proximity Identifiers, and neither is it informed if a new one has been collected by the framework. The only output of the ENF upon an infection is a collection of temporary exposure keys as shown in *Figure 11*. Those are subsequently called diagnosis keys.
```
Comment (Ein-Tim) [inline solution_architecture.md:234]: Suggestion: Add the information that the app also is not informed if these keys are deleted. This is quite an important info and answers questions like: "Why doesn't the app change if I delete the keys via OS settings?"

File: solution_architecture.md [hunk 3]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
 
-### Attenuation Buckets
+### Risk Calculation
 
-While in the first version of the Exposure Notification Framework, Apple and Google allowed to define multiple thresholds for the attenuation, this functionality can now be implemented within the application through the data from the exposure windows.
+The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk _red_. The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolling Proximity Identifier, the Temporary Exposure Key or the exact time) remains within the secure storage of the framework and cannot be retrieved by the application.
```
Comment (Ein-Tim) [inline solution_architecture.md:264]: Add brackets to the word "red".

```suggestion
The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk (_red_). The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolling Proximity Identifier, the Temporary Exposure Key or the exact time) remains within the secure storage of the framework and cannot be retrieved by the application.
```

File: solution_architecture.md [hunk 4]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
+    <details>
+    <summary>Figure 17: Determine DSOS for 'No information'</summary>
+
+    | ![Figure 17: Determine DSOS for 'No information'](images/solution_architecture/dsos-no-information.png "Figure 17: Determine DSOS for 'No information'") |
+    |:--:|
+    | **Figure 17: Determine DSOS for 'No information'**|
+
+    </details>
+
+2. **No symptoms** - the user indicates that he/she has no symptoms
```
Comment (Ein-Tim) [inline solution_architecture.md:306]: Use they to be gender neutral.

```suggestion
2. **No symptoms** - the user indicates that they have no symptoms
```

File: solution_architecture.md [hunk 5]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
+    <details>
+    <summary>Figure 18: Determine DSOS for 'No symptoms'</summary>
+
+    | ![Figure 18: Determine DSOS for 'No symptoms'](images/solution_architecture/dsos-no-symptoms.png "Figure 18: Determine DSOS for 'No symptoms'") |
+    |:--:|
+    | **Figure 18: Determine DSOS for 'No symptoms'**|
+
+    </details>
+
+3. **No date information** - the user indicates that he/she has symptoms but declines to provide any date information
```
Comment (Ein-Tim) [inline solution_architecture.md:317]:
```suggestion
3. **No date information** - the user indicates that they have symptoms but decline to provide any date information
```

File: solution_architecture.md [hunk 6]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
+    <details>
+    <summary>Figure 19: Determine DSOS for 'No date information'</summary>
+
+    | ![Figure 19: Determine DSOS for 'No date information'](images/solution_architecture/dsos-date-unknown.png "Figure 19: Determine DSOS for 'No date information'") |
+    |:--:|
+    | **Figure 19: Determine DSOS for 'No date information'**|
+
+    </details>
+
+4. **Date range** - the user indicates that he/she has symptoms and provides a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")
```
Comment (Ein-Tim) [inline solution_architecture.md:328]:
```suggestion
4. **Date range** - the user indicates that they have symptoms and provide a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")
```

File: solution_architecture.md [hunk 7]
```diff
@@ -251,41 +255,108 @@ All exposure events are collected by the ENF internally and are split up into "E
...
+    <details>
+    <summary>Figure 20: Determine DSOS for 'Date range'</summary>
+
+    | ![Figure 20: Determine DSOS for 'Date range'](images/solution_architecture/dsos-date-range.png "Figure 20: Determine DSOS for 'Date range'") |
+    |:--:|
+    | **Figure 20: Determine DSOS for 'Date range'**|
+
+    </details>
+
+5. **Specific date** - the user indicates that he/she has symptoms and provides a specific date
```
Comment (Ein-Tim) [inline solution_architecture.md:339]:
```suggestion
5. **Specific date** - the user indicates that they have symptoms and provide a specific date
```

**Review by Ein-Tim — COMMENTED** [review]
Some suggestions, e.g. to use "they" instead of "he/she". but in general: Looks good to me!

File: solution_architecture.md (hunk 4 above)
Comment (mlenkeit) [reply solution_architecture.md:306]: @Ein-Tim I like the proposal of dropping _he/she_ but I'm not sure if _they_ is such a good replacement. How about just dropping the relative clause and rephrase it to _the user has no symptoms_?

File: solution_architecture.md (hunk 2 above)
Comment (mlenkeit) [reply solution_architecture.md]: Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/0e19cdfc1325f4af10322efb5237cc471cef2750.

File: solution_architecture.md (hunk 4 above)
Comment (Ein-Tim) [reply solution_architecture.md:306]: Sounds good!

Comment (mlenkeit) [reply solution_architecture.md]: fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/19341c3fef9edf172d14c7773f933263b71ddefc

**Review by Ein-Tim — APPROVED** [review]
LGTM!
